### DISCUSSION TITLE
Acquire Token Interactive issue in docker container

### DISCUSSION THREAD

**Discussion by bruteforcesid** [AzureAD/microsoft-authentication-library-for-python#736] [Q&A] [answered]
Hello,

I have an azure web app(x) deployed via docker container. I have another microsoft entra app with client id y, of which i want to acquire token by interactive method from user. 

Code used in x app(written in python) - 

app = msal.PublicClientApplication(
client_id = y,
authority=authority,
allow_broker=True     //Optional
)

result = app.acquire_token_interactive(scopes=scopes, prompt="select_account", parent_window_handle=msal.PublicClientApplication.CONSOLE_WINDOW_HANDLE //optional with allow broker
)

While this is working fine on local environment without docker, but while running with docker, getting "Found no browser in the current environment. If this....."

Got similar issue with https://github.com/AzureAD/microsoft-authentication-library-for-python/issues/422 but since that is an old thread, not sure if we have got something for this case. Even if some manual way or workaround is there, please let us know.

**Comment by rayluo** [comment, marked as answer]
Per offline conversation, the "web app X" in question is an Chainlit application.

1. MSAL's `PublicClientApplication.acquire_token_interactive()` can be used by a desktop script which is potentially running inside docker [if the container was started by a certain ways](https://github.com/AzureAD/microsoft-authentication-library-for-python/blob/1.30.0/msal/oauth2cli/authcode.py#L306-L311). However, this API is not designed to be used by a web app. And web app shall not use it.
2. We have other samples tailored for web app (or specifically, tailored for some popular web frameworks, for example, this [sample for Flask](https://learn.microsoft.com/en-us/entra/identity-platform/quickstart-web-app-python-flask?tabs=windows)), but we do not currently have a sample for Chainlit.
3. [Chainlit itself apparently has built-in AAD auth integration](https://docs.chainlit.io/authentication/oauth#azure-active-directory).

So, you shall go with `#3`.
